test/remote-github-patch-text.ts

```diff
@@ -0,0 +1,454 @@
+          contents_url:
+            'https://api.github.com/repos/TomKristie/HelloWorld/contents/Readme.md?ref=eb53f3871f56e8dd6321e44621fe6ac2da1bc120',
+          patch:
+            '@@ -1,2 +1,5 @@\n Hello world\n-!\n+Goodbye World\n+gOodBYE world\n+\n+Goodbye World',
+        },
+        {
+          sha: '8b137891791fe96927ad78e64b0aad7bded08bdc',
+          filename: 'foo/foo.txt',
+          status: 'modified',
+          additions: 1,
+          deletions: 1,
+          changes: 2,
+          blob_url:
+            'https://github.com/TomKristie/HelloWorld/blob/eb53f3871f56e8dd6321e44621fe6ac2da1bc120/foo/foo.txt',
+          raw_url:
+            'https://github.com/TomKristie/HelloWorld/raw/eb53f3871f56e8dd6321e44621fe6ac2da1bc120/foo/foo.txt',
+          contents_url:
+            'https://api.github.com/repos/TomKristie/HelloWorld/contents/foo/foo.txt?ref=eb53f3871f56e8dd6321e44621fe6ac2da1bc120',
+          patch: '@@ -1 +1 @@\n-Hello foo\n+',
+        },
+        {
+          sha: '3b18e512dba79e4c8300dd08aeb37f8e728b8dad',
+          filename: 'helloworld.txt',
+          status: 'removed',
+          additions: 0,
+          deletions: 1,
+          changes: 1,
+          blob_url:
+            'https://github.com/TomKristie/HelloWorld/blob/f5da827a725a701302da7db2da16b1678f52fdcc/helloworld.txt',
+          raw_url:
+            'https://github.com/TomKristie/HelloWorld/raw/f5da827a725a701302da7db2da16b1678f52fdcc/helloworld.txt',
+          contents_url:
+            'https://api.github.com/repos/TomKristie/HelloWorld/contents/helloworld.txt?ref=f5da827a725a701302da7db2da16b1678f52fdcc',
+          patch: '@@ -1 +0,0 @@\n-hello world',
+        },
+      ],
+    };
+    sandbox.stub(octokit.pulls, 'listFiles').resolves(listFilesOfPRResult);
+
+    // tests
+    const {patches, filesMissingPatch} = await getCurrentPulLRequestPatches(
+      octokit,
+      upstream,
+      pullNumber,
+      pageSize
+    );
+    expect(patches.size).equals(3);
+    expect(patches.get(listFilesOfPRResult.data[0].filename)).equals(
+      listFilesOfPRResult.data[0].patch
+    );
+    expect(patches.get(listFilesOfPRResult.data[1].filename)).equals(
+      listFilesOfPRResult.data[1].patch
+    );
+    expect(patches.get(listFilesOfPRResult.data[2].filename)).equals(
+      listFilesOfPRResult.data[2].patch
+    );
+    expect(filesMissingPatch.length).equals(0);
+  });
+  it('Passes the error message up from octokit when octokit fails', async () => {
+    // setup
+    const errorMsg = 'Error message';
+    sandbox.stub(octokit.pulls, 'listFiles').rejects(Error(errorMsg));
+    try {
+      await getCurrentPulLRequestPatches(
+        octokit,
+        upstream,
+        pullNumber,
+        pageSize
+      );
+      expect.fail(
+        'The getCurrentPulLRequestPatches function should have failed because Octokit failed.'
+      );
+    } catch (err) {
+      expect(err.message).to.equal(errorMsg);
```

Comment:
Testing error messages is a code smell that we're not throwing the right kind of error. Although maybe this is a javascript thing that's unavoidable?

Reply:
I am stubbing octokit which is called by a function I am actually testing. I am testing that when octokit fails, the caller rethrow's octokit's error. What I am doing here is creating a fake error for the fake Octokit method. If we went with a more accurate error, that would only affect the scope of the tests, not the runtime.